src/jit/morph.cpp

```diff
@@ -4354,32 +4476,113 @@ void Compiler::fgMorphMultiregStructArgs(GenTreeCall* call)
 //    For the GT_OBJ case will clone the address expression and generate two (or more)
 //    indirections.
 //    Currently the implementation only handles ARM64 and will NYI for other architectures.
-//    And for ARM64 we do not ye handle HFA arguments, so only 16-byte struct sizes are supported.
 //
 GenTreePtr    Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPtr fgEntryPtr)
 {
-    GenTreeArgList*  newArg = nullptr;
     assert(arg->TypeGet() == TYP_STRUCT);
-    GenTreePtr argValue = arg;
 
 #ifndef _TARGET_ARM64_
-    NYI("fgMorphMultiregStructArg non-ARM64 implementation");
+    NYI("fgMorphMultiregStructArg requires implementation for this target");
 #endif
 
-    // If we have a GT_OBJ of a GT_ADDR then
-    //  we set argValue to the child node ofthe GT_ADDR
+#if FEATURE_MULTIREG_ARGS
+    // Examine 'arg' and setup argClass 
+    //
+    CORINFO_CLASS_HANDLE objClass = nullptr;
+    GenTreePtr           argValue = arg;   // normally argValue will be arg, but see right below
     if (arg->OperGet() == GT_OBJ)
     {
-        GenTreePtr argAddr = arg->gtOp.gtOp1;
+        GenTreeObj*  argObj = arg->AsObj();
+        objClass = argObj->gtClass;
+
+        // If we have a GT_OBJ of a GT_ADDR then we set argValue to the child node of the GT_ADDR
+        //
+        if (argObj->gtOp1->OperGet() == GT_ADDR)
+        {
+            argValue = argObj->gtOp1->gtOp.gtOp1;
+        }
+    }
+    else if (arg->OperGet() == GT_LCL_VAR)
+    {
+        GenTreeLclVarCommon* varNode = arg->AsLclVarCommon();
+        unsigned varNum = varNode->gtLclNum;
+        assert(varNum < lvaCount);
+        objClass = lvaGetStruct(varNum);
+    }
+    noway_assert(objClass != nullptr);
 
-        if (argAddr->OperGet() == GT_ADDR)
+    unsigned   structSize = info.compCompHnd->getClassSize(objClass);
```

Comment:
if arg is a GT_LCL_VAR we can get its size using lvSize() and avoid Jit-EE interface call?